src/hstring.rs

```diff
@@ -65,6 +77,27 @@ impl HString {
 
         self.ptr = std::ptr::null_mut();
     }
+
+    /// # Safety
+    /// len must not be less than the number of items in the iterator.
+    unsafe fn from_wide_iter<I: Iterator<Item = u16>>(iter: I, len: u32) -> HString {
+        if len == 0 {
```

Comment:
Since the safety invariant is easy to test for, I think we should have an `assert_debug!` that checks that the invariant is respected. This would only get triggered if we've made a mistake somewhere in winrt-rs (since this is a private function), but I think the small runtime hit on debug builds is worth it.

Reply:
Ok sounds good I'll put a `debug_assert!((index as u32) < len);` in the loop. 
Also, shouldn't `Header::allocate` allocate `len + 1` for the null terminator?

Reply:
`Header::alloc` already accounts for the terminator.